lib/r10k/api.rb

```diff
@@ -88,72 +91,78 @@ def parse_puppetfile(io_or_content)
         parser.instance_eval(io_or_content)
       end
 
-      return { :modules => builder.build }
+      return builder.build
     end
 
     # Create an unresolved environment hashmap representing the desired state of a Puppet environment based on a source and branch.
     #
-    # @param source [Hash] A hashmap representing a control repo source (like would be defined in r10k.yaml)
+    # @param control_source [Hash] A hashmap representing a control repo source (like would be defined in r10k.yaml)
     # @param env_name [String] Environment name to build an envmap for.
-    # @param base_cachedir [String] Root of where r10k is caching things.
+    # @option opts [String] :cachedir Path where r10k should cache things.
     # @return [Hash] A hashmap representing the desired state of the environment matching the given branch in the given source.
     # @raise [NotImplementedError] Control repo source has a type that is not currently supported.
-    # @raise [RuntimeError] Something bad happened, see exception message.
-    def envmap_from_source(source, env_name, base_cachedir)
+    # @raise [RuntimeError]
+    def envmap_from_source(control_source, env_name, opts={})
       # TODO: enforce valid Puppet environment name here? verify prefix? maybe just use R10K::Environment::Name?
       branch_name = env_name
 
-      if source[:prefix]
-        if source[:prefix].is_a? String
-          branch_name = env_name.gsub(/^#{Regexp.quote(source[:prefix])}_/, '')
+      if control_source[:prefix]
+        if control_source[:prefix].is_a? String
+          branch_name = env_name.gsub(/^#{Regexp.quote(control_source[:prefix])}_/, '')
         else
-          branch_name = env_name.gsub(/^#{Regexp.quote(source[:name])}_/, '')
+          branch_name = env_name.gsub(/^#{Regexp.quote(control_source[:name])}_/, '')
         end
       end
 
-      case source[:type].to_sym
+      case control_source[:type].to_sym
       when :git
-        cachedir = cachedir_for_git_remote(source[:remote], base_cachedir)
+        git_dir = cachedir_for_git_remote(control_source[:remote], opts[:cachedir])
 
         begin
-          version = git.rev_parse(branch_name, git_dir: cachedir)
+          commit_sha = git.resolve_commit(git_dir, branch_name)
         rescue R10K::Git::GitError => e
           raise RuntimeError.new("Unable to resolve branch name '#{branch_name}' to a Git commit: #{e.message}")
         end
 
         # TODO: figure out how to pass through base_path option
-        puppetfile = get_puppetfile(:git, cachedir, version)
+        puppetfile = get_puppetfile(source version)
```

Comment:
kind of surprised that's not a parse error, maybe the commas are optional in Ruby?